console: Add Azure & Keycloak docs (#1429)
redpanda-data/documentation: docs/manage/security/console/azure-ad.mdx

---
title: Azure AD SSO Setup
---

<head>
    <meta name="title" content="Azure AD SSO Authentication in Redpanda Console | Redpanda Docs"/>
    <meta name="description" content="Configure authentication with external identity providers such as Google, GitHub or Okta in Redpanda Console."/>
</head>

:::info
This feature requires an [Enterprise license](../../../../get-started/licenses). To upgrade, contact [Redpanda sales](https://redpanda.com/try-redpanda?section=enterprise-cloud).
:::

> Integrating Redpanda Console with Azure AD allows your users to use their Azure AD identities to sign-in to Console.

Reviewer: ```suggestion
Integrating Redpanda Console with Azure AD allows your users to use their Azure AD identities to sign in to Console.
```

Author: This comment is also copied and I think this dash had been removed in a previous PR. Whatever is the right version should go into all pages I assume?

Reviewer 2: Looks like we need to improve the process for docs reviewing these other source pages. Keeping mistakes/writing errors because they appear in other docs and we don't want them to be inconsistent is not a good approach. In other words, just because the IpD docs have errors does not mean ours should...

Reviewer: @Feediver1 
Martin is saying that he used the same wording in our docs for each IdP, so any mistakes were propagated to all our IdP documentation. I can file an issue to make all our SSO docs follow the same basic outline, and I can point out that this error needs to be fixed.

This guide assumes you already have an Azure AD account and permissions to create Applications within your directory.

## Create an OpenID connect application

TODO: Write text, possibly refer to: https://learn.microsoft.com/en-us/power-apps/maker/portals/configure/configure-openid-settings

:::note

> Below configurations assume that you want to host Redpanda Console so that it would be accessible via

Reviewer: ```suggestion
The following configurations assume that you want to host Redpanda Console so that it would be accessible from
```
Another idea:
The following configurations assume that you want Redpanda Console to be accessible from

Reviewer 2: A configuration cannot make an assumption. A configuration can be based on an assumption. So suggesting:

The following configurations are based on the assumption that you want to host Redpanda Console so that it is accessible from...

Reviewer: Shortened to:
The following configurations are based on hosting Redpanda Console so it is accessible from
`https://console.<your-company>.com`.

`https://console.yourcompany.com`.
:::

> - **Name:** Any descriptive name for your specific Console deployment (for example Console Analytics Prod)

Reviewer: For consistency, use the imperative:
```suggestion
- **Name:** Provide a descriptive name for your specific Console deployment (for example, Console Analytics Prod)
```

    # "OpenID Connect metadata document" contains the 
    # provider URL, however you must trim the suffix 
    # "/.well-known/openid-configuration" so that it 
    # matches the expected format as shown above.
    providerUrl: ""

    clientId: ""
    clientSecret: ""

    #  userIdentifyingClaimKey is a relevant property if you want 
    #  to use a different claim key to identify users in the 
    #  role binding. By default, we will use the 'oid' claim key,
    #  which resolves to the unique user ID within the identity 
    #  provider. This means that you must provide the oid 
    #  identifier in the roleBindings as 'name' as well. 
    #  Other common options are:
    #  - upn (unique principal name - you need to add the upn 
    #    claim as a claim for id tokens in your Azure AD application)
    #  - email (under certain conditions there's no value for 
    #    the email propagated)
    userIdentifyingClaimKey: "oid"

    # The directory config is only required if you want to use 
    # Azure AD groups in your role bindings. Described further
    # in the next section.
    # directory:
    #   tenantId: ""
```

## RBAC Azure AD groups sync

> If you want to bind roles to Azure groups from your organization you have to provide the `tenantId` in the
> directory configuration and add API permissions to your client application. To retrieve the `tenantId` go to
> your registered application in the Azure Active Directory portal. The "Directory (tenant) ID" is listed

Reviewer: ```suggestion
To bind roles to Azure groups from your organization, provide the `tenantId` in the
directory configuration and add API permissions to your client application. To retrieve the `tenantId`, go to
your registered application in the Azure Active Directory portal. The "Directory (tenant) ID" is listed
```

> If you want to bind roles to Azure groups from your organization you have to provide the `tenantId` in the
> directory configuration and add API permissions to your client application. To retrieve the `tenantId` go to
> your registered application in the Azure Active Directory portal. The "Directory (tenant) ID" is listed
> along other configurations in the essentials section.

Reviewer: ```suggestion
To bind roles to Azure groups from your organization, provide the `tenantId` in the
directory configuration and add API permissions to your client application. To retrieve the `tenantId`, go to
your registered application in the Azure Active Directory portal. The "Directory (tenant) ID" is listed along with other configurations in the essentials section.
```

> If you want to bind roles to Azure groups from your organization you have to provide the `tenantId` in the
> directory configuration and add API permissions to your client application. To retrieve the `tenantId` go to
> your registered application in the Azure Active Directory portal. The "Directory (tenant) ID" is listed
> along other configurations in the essentials section.

Reviewer: ```suggestion
To bind roles to Azure groups from your organization, provide the `tenantId` in the
directory configuration and add API permissions to your client application. To retrieve the `tenantId`, go to
your registered application in the Azure Active Directory portal. The "Directory (tenant) ID" is listed
along with other configurations in the essentials section.
```

Reviewer: Done.

---
title: Azure AD SSO Setup
---

<head>
    <meta name="title" content="Azure AD SSO Authentication in Redpanda Console | Redpanda Docs"/>
    <meta name="description" content="Configure authentication with external identity providers in Redpanda Console."/>
    <link rel="canonical" href="https://docs.redpanda.com/docs/manage/security/console/azure-ad" />
</head>

:::info
This feature requires an [Enterprise license](../../../../get-started/licenses). To upgrade, contact [Redpanda sales](https://redpanda.com/try-redpanda?section=enterprise-cloud).
:::

By integrating Redpanda Console with Azure AD, your users can sign in to Redpanda Console using their Azure AD login credentials.

## Prerequisites

You must have:

- An [Azure AD account](https://learn.microsoft.com/en-us/azure/active-directory/fundamentals/sign-up-organization) and permissions to create applications within your directory.

> - An OpenID Connect (OIDC) provider](https://learn.microsoft.com/en-us/power-apps/maker/portals/configure/configure-openid-settings) configured for your Azure AD account.

Reviewer 2: ```suggestion
- An [OpenID Connect (OIDC) provider](https://learn.microsoft.com/en-us/power-apps/maker/portals/configure/configure-openid-settings) configured for your Azure AD account.
```

Reviewer 2: ^^ Link wan't rendering correctly

  :::note

>   Set the redirect URI to the domain where Redpanda Console is hosted followed by the `/login/callbacks/azure-ad` path. For example, `https://console.<your-company>.com/login/callbacks/azure-ad`

Reviewer 3: ```suggestion
  Set the redirect URI to the domain where Redpanda Console is hosted followed by the `/login/callbacks/azure-ad` path. For example, `https://console.<your-company>.com/login/callbacks/azure-ad`.
```

### Generate the client secret

Redpanda Console requires a client secret to validate the connection request from the redirect URI. To generate a client secret:

1. In the **Essentials** section of the summary page, click **Add a certificate or secret** next to **Client credentials**.

2. On the **Certificates & secrets** page, select the **Client secrets** tab.

3. Click **New client secret**.

4. In the **Add a client secret** pane, enter a description that identifies the app associated with the client secret.

5. Choose a time period that determines when the client secret will expire; for example, 180 days.

When the client secret is generated successfully, it appears under the **Client secrets** tab with its description, expiration date, value, and secret ID.

:::important
Copy the client secret value into the `redpanda-console-config.yaml` file before you leave the **Certificates and secrets** page. After you navigate away from the page, only the first few characters of the client secret value are displayed and you won't be able to copy the value into the configuration file.
:::

## Edit the console configuration file

> Edit the `redpanda-console-config.yaml` file, which is in `/etc/redpanda`, and incorporate the details from your app registration.

Author: Yes users would provide the configurations (not only Redpanda Console but also other configs) via the `values.yaml`

> Edit the `redpanda-console-config.yaml` file, which is in `/etc/redpanda`, and incorporate the details from your app registration.

Reviewer: What do you think of this rewrite:
>Edit the console configuration file associated with your deployment method and incorporate the details from your client application. For example, Kubernetes deployments use the `values.yaml` file. Linux deployments use the `redpanda-console-config.yaml` file, which is in `/etc/redpanda`.

I'm not sure if our users need that much guidance, but it seemed more complete to include that information. If you think this is unnecessary, what about:
>Edit the console configuration file associated with your deployment method and incorporate the details from your client application.

> Edit the `redpanda-console-config.yaml` file, which is in `/etc/redpanda`, and incorporate the details from your app registration.

Author: Both is fine for me.

> Edit the `redpanda-console-config.yaml` file, which is in `/etc/redpanda`, and incorporate the details from your app registration.

Reviewer: I went with the first rewrite, which has more detail. 
@weeco  and @Feediver1 - Which do you prefer?

By integrating Redpanda Console with Azure AD, your users can sign in to Redpanda Console using their Azure AD login credentials.

## Prerequisites

You must have:

- An [Azure AD account](https://learn.microsoft.com/en-us/azure/active-directory/fundamentals/sign-up-organization) and permissions to create applications within your directory.

> - An OpenID Connect (OIDC) provider](https://learn.microsoft.com/en-us/power-apps/maker/portals/configure/configure-openid-settings) configured for your Azure AD account.

Reviewer 3: I've added a note below with the requirements for the redirect URI.

- An [Azure AD account](https://learn.microsoft.com/en-us/azure/active-directory/fundamentals/sign-up-organization) and permissions to create applications within your directory.

### Register your app
- An OpenID Connect (OIDC) provider](https://learn.microsoft.com/en-us/power-apps/maker/portals/configure/configure-openid-settings) configured for your Azure AD account.

To register your application, log in to the Azure Portal and provide the following information:
## Create an OpenID connect application

* **Display name** - Choose a descriptive name for your specific Redpanda Console deployment (for example Console Analytics Prod)
* **Supported account types** - Select the types of accounts that can access the API (choose single tenant, multi-tenant, or personal accounts). If you are unsure, click "Help me choose...".

> * **Redirect URIs** Select Web as the platform and enter `https://console.<your-company>.com/login/callbacks/azure-ad` in the text box. Replace `<your-company>.com` with your company's domain name.

Author: Where did that go? This is required information for the user. If they don't know the pattern for the callback url they would need to look through the code to find it and that is closed source.

Reviewer 3: I've kept the information about the redirect URI.

I've removed steps that document third-party tools because it's a best practice. We cannot guarantee that any steps we provide for third-party tools remain up to date.

:::

After the app is registered, a summary page opens. This page displays a list of **Essentials**, which you will need when you edit the `redpanda-console-config.yaml file`.

For detailed instructions, see the Microsoft documentation for [configuring an OpenID connect provider](https://learn.microsoft.com/en-us/power-apps/maker/portals/configure/configure-openid-settings).

### Generate the client secret

Redpanda Console requires a client secret to validate the connection request from the redirect URI. To generate a client secret:

1. In the **Essentials** section of the summary page, click **Add a certificate or secret** next to **Client credentials**.

2. On the **Certificates & secrets** page, select the **Client secrets** tab.

3. Click **New client secret**.

4. In the **Add a client secret** pane, enter a description that identifies the app associated with the client secret.

5. Choose a time period that determines when the client secret will expire; for example, 180 days.

When the client secret is generated successfully, it appears under the **Client secrets** tab with its description, expiration date, value, and secret ID.

> 

Author: Why is this removed?

Reviewer 3: I've removed steps that document third-party tools because it's a best practice. We cannot guarantee that any steps we provide for third-party tools remain up to date.

Instead, we should provide links and any Redpanda-specific input requirements.
